v2/logging.go

```diff
@@ -1,45 +1,23 @@
 package shuttle
 
 import (
-	"context"
-	"fmt"
-	"os"
-	"time"
+	"io"
+	"log/slog"
 )
 
-type Logger interface {
-	Info(s string)
-	Warn(s string)
```

Comment:
these funcs don't take context.
2 ways around it: 
- this is used internally, but implemented by consumers. we can create a new ContextLogger interface that take in the ctx, and use the ContextLogger everywhere in goshuttle
- make getLogger take in a ctx, and use that context in the internal calls. that requires a bit of care on our side (to get a new logger when needed, and not reuse one when the context is different) but I think that's quite easy in go-shuttle